### DISCUSSION TITLE
Beginner connecting with QR Code from Swift App

### DISCUSSION THREAD

**Discussion by quangas** [WalletConnect/walletconnect-monorepo#3504] [Developer Support]
I had a read of this link:

https://medium.com/walletconnect/beginner-guide-to-walletconnect-v2-0-for-ios-developers-32284178da18

In this article it shows the QR code to scan but it doesn't show how to generate this.

My aim is to generate a QR code from a React site, and then from the iOS side the phone will scan this QR code and then establish a connection. The mobile will sign/verify the connection, and then on the React site it will acknowledge this sign/verify and the React site can start enabling web3 features. The QR code between the React site and the mobile app acts like a handshake. Is this possible?

Please send over some docs or guides to show how this can be done. i.e. do you use the @walletconnect/ethereum-provider to generate the QR code? how do you establish/communicate with the session.

**Comment by glitch-txs** [comment]
Hi, do you want to build a website or a wallet? or both?
here our docs: https://docs.walletconnect.com/2.0/
